Week1/homework/app.js

```diff
@@ -1,11 +1,171 @@
+    the_da_vinci_code: {
+      title: 'The Da Vinci Code',
+      language: 'english',
+      author: 'Dan Brown',
+    },
+    think_and_grow_rich: {
+      title: 'Think and Grow Rich',
+      language: 'English',
+      author: 'Napoleon Hill',
+    },
+    gone_with_the_wind: {
+      title: 'Gone With The Wind',
+      language: 'English',
+      author: 'Margaret Mitchell',
+    },
+    peygamberim: {
+      title: 'Peygamberim',
+      language: 'Turkish',
+      author: 'Salih Suruc',
+    },
+    benim_adim_kirmizi: {
+      title: 'Benim Adim Kirmizi',
+      language: 'Turkish',
+      author: 'Orhan Pamuk',
+    },
+  };
+
+  // books infos appended in ul as li
+  function copyMyBooksInfoToHtml() {
+    const h1 = document.createElement('h1');
+    h1.innerText = 'MY BOOKS';
+    document.body.appendChild(h1);
+    const ul = document.createElement('ul');
+    for (const book in myBooksInfo) {
+      if (Object.prototype.hasOwnProperty.call(myBooksInfo, book)) {
+        const li = document.createElement('li');
+        const h2 = document.createElement('h2');
+        const authorP = document.createElement('p');
+        const languageP = document.createElement('p');
+        h2.innerHTML = myBooksInfo[book].title;
+        authorP.innerHTML = 'Author: ' + myBooksInfo[book].author;
+        languageP.innerHTML = 'Original Language: ' + myBooksInfo[book].language;
+        li.id = book.slice(0, 6);
+        li.appendChild(h2);
+        li.appendChild(authorP);
+        li.appendChild(languageP);
+        ul.appendChild(li);
+        document.body.appendChild(ul);
+      }
+    }
+  }
+  copyMyBooksInfoToHtml();
+
+  const myBooksCovers = {
+    huzursuzluk: {
+      source: 'huzursuzluk.jpg',
+    },
+    oliver_twist: {
+      source: 'oliver_twist.jpg',
+    },
+    war_and_peace: {
+      source: 'war_and_peace.jpg',
+    },
+    the_death_of_ivan_ilyich: {
+      source: 'the_death_of_ivan_ilyich.jpg',
+    },
+    resurrection: {
+      source: 'resurrection.jpg',
+    },
+    the_da_vinci_code: {
+      source: 'the_da_vinci_code.jpg',
+    },
+    think_and_grow_rich: {
+      source: 'think_and_grow_rich.jpg',
+    },
+    gone_with_the_wind: {
+      source: 'gone_with_the_wind.jpg',
+    },
+    peygamberim: {
+      source: 'peygamberim.jpg',
+    },
+    benim_adim_kirmizi: {
+      source: 'benim_adim_kirmizi.jpg',
+    },
+  };
+
+  // in next function;
+  // IDs of li elements stored in 'idList'
+  // URLs of images stored in 'urlList'
+  // 250px height set for images
+  function putCoversToRightLiElem() {
+    const urlList = [];
+    for (const i in myBooksCovers) {
+      if (Object.prototype.hasOwnProperty.call(myBooksCovers, i)) {
+        urlList.push(i + '.jpg');
+      }
+    }
+    const idList = [];
+    for (const i in myBooks) {
+      if (Object.prototype.hasOwnProperty.call(myBooks, i)) {
+        idList.push(myBooks[i].slice(0, 6));
+      }
+    }
+
+    for (let i = 0; i < urlList.length; i++) {
```

Comment:
Instead of turning everything in to lists you can get the right url from the `myBooksCovers` like this:
```
for(const myBook in myBooks) {
    const myBookCovers = myBooksCovers[myBook].source;
    // .... rest of the code that creates the image and appends it to the right li element
}
```

Reply:
yes you're right ! there are 'url's already. I need to take from there.
Unfortunately I did not understand your code above.
I create a new variable inside for in loop? and this is instead of urlList?
I tried just it doesn't work for now ""TypeError: undefined is not an object (evaluating 'myBooksCovers[myBook].source')""

Reply:
its ok. I solved that. and I changed my silly code(... + '.jpg') :)
thanks a lot for your valuable time.